chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/certifiedproduct/UpdateActionPermissions.java

```diff
@@ -1,25 +1,38 @@
 package gov.healthit.chpl.permissions.domains.certifiedproduct;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.springframework.stereotype.Component;
 
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.domain.ListingUpdateRequest;
 import gov.healthit.chpl.permissions.domains.ActionPermissions;
 
 @Component("certifiedProductUpdateActionPermissions")
 public class UpdateActionPermissions extends ActionPermissions {
+    private static final Logger LOGGER = LogManager.getLogger(UpdateActionPermissions.class);
 
     @Override
     public boolean hasAccess() {
         return false;
     }
 
     @Override
-    public boolean hasAccess(Object obj) {
-        if (!(obj instanceof Long)) {
+    public boolean hasAccess(final Object obj) {
+        if (!(obj instanceof ListingUpdateRequest)) {
             return false;
         } else if (getResourcePermissions().isUserRoleAdmin() || getResourcePermissions().isUserRoleOnc()) {
             return true;
         } else if (getResourcePermissions().isUserRoleAcbAdmin()) {
-            Long acbId = (Long) obj;
+            ListingUpdateRequest updateRequest = (ListingUpdateRequest) obj;
```

Comment:
This is the part where we're going to have merge issues with the ticket I have in "Ready for Review", right?

Reply:
Yes... I had made changes all over place by the time we talked about it, but whenever we end up merging this we should just keep the code from your ticket.